packages/analytics/src/providers/kinesis/utils/getEventBuffer.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { KinesisBufferEvent, KinesisEventBufferConfig } from '../types';
+import {
+	EventBuffer,
+	groupBy,
+	IAnalyticsClient,
+	recordToTupleList,
+} from '../../../utils';
+import { KinesisClient, PutRecordsCommand } from '@aws-sdk/client-kinesis';
+
+const eventBufferMap: Record<string, EventBuffer<KinesisBufferEvent>> = {};
+const cachedClients: Record<string, KinesisClient> = {};
+
+const createKinesisPutRecordsCommand = (
+	streamName: string,
+	events: KinesisBufferEvent[]
+): PutRecordsCommand =>
+	new PutRecordsCommand({
+		StreamName: streamName,
+		Records: events.map(x => ({
+			PartitionKey: x.partitionKey,
+			Data: x.event,
+		})),
+	});
+
+const submitEvents = async (
+	events: KinesisBufferEvent[],
+	client: KinesisClient,
+	resendLimit?: number
+): Promise<KinesisBufferEvent[]> => {
+	const groupedByStreamName = recordToTupleList(
+		groupBy(x => x.streamName, events)
+	);
+	const requests = groupedByStreamName
+		.map(x => createKinesisPutRecordsCommand(...x))
```

Comment:
Nit, but ditto for other occurrences of `x` just to make this a little more readable

```suggestion
		.map(events => createKinesisPutRecordsCommand(...events))
```

Reply:
will update it